qutip/qip/circuit.py

```diff
@@ -414,14 +414,16 @@ def remove_gate(self, index=None, end=None, name=None, remove="first"):
         remove : string
             If first or all gate are to be removed.
         """
-        if index is not None and index <= self.N:
-            if end is not None and end <= self.N:
+        if index is not None and index <= len(self.gates):
+            if end is not None and end <= len(self.gates):
                 for i in range(end - index):
                     self.gates.pop(index + i)
             elif end is not None and end > self.N:
                 raise ValueError("End target exceeds number of gates.")
             else:
                 self.gates.pop(index)
+        else:
+            raise ValueError("Index exceeds number of gates.")
 
         elif name is not None and remove == "first":
             for gate in self.gates:
```

Comment:
While they do do it in a somewhat obscure manner, the CI errors are telling you that there's invalid syntax here; you've inserted an `else` statement before an `elif`.

Reply:
Sorry, my bad. I've fixed the problem

Reply:
Sorry, my bad. It should work now